nevergrad/functions/stsp/core.py

```diff
@@ -11,17 +11,18 @@
 
 import matplotlib.pyplot as plt
 import numpy as np
-from ... import instrumentation as inst
+from nevergrad import instrumentation as inst
+from ..base import ExperimentFunction
 
 
-class STSP(inst.InstrumentedFunction):  # TODO avoid seeding functions :s (in future ParametrizedFunction)
+class STSP(ExperimentFunction):
 
-    def __init__(self, seed: int = 0, dimension: int = 500) -> None:
+    def __init__(self, dimension: int = 500) -> None:
         super().__init__(self._simulate_stsp, inst.var.Array(dimension))
-        self.order = np.arange(0, self.instrumentation.dimension)
-        self.instrumentation.random_state.seed(seed)
-        self.x = self.instrumentation.random_state.normal(size=self.dimension)
-        self.y = self.instrumentation.random_state.normal(size=self.dimension)
+        self.register_initialization(dimension=dimension)
+        self.order = np.arange(0, self.dimension)
+        self.x = self.parametrization.random_state.normal(size=self.dimension)
```

Comment:
just to make it clear once, `instrumentation` becomes `parametrization` in `ExperimentFunction` for forward compatibility with future updates (see #391 ), but the underlying instance is still an `Instrumentation`. At some point in the near future, this will also be the case for optimizers (but after the merge of new parametrization framework).